File: solutions/1164-missing-number-in-arithmetic-progression/missing-number-in-arithmetic-progression.java
class Solution {
    public int missingNumber(int[] arr) {
        int d = arr[1] - arr[0];
        if(d < 0){
            d = Math.max(arr[2] - arr[1], d);
        }else{
            d = Math.min(arr[2] - arr[1], d);
        }

        for(int i=0; i<arr.length -1; i++){
            if(arr[i] + d != arr[i+1]){
                return arr[i] + d;
            }
        }

        return arr[1];
    }
}

File: solutions/1164-missing-number-in-arithmetic-progression/missing-number-in-arithmetic-progression.py
class Solution:
    def missingNumber(self, arr: List[int]) -> int:
        n = len(arr) + 1
        a = arr[0]
        b = arr[-1]
        d = (b - a) / (n - 1)

        cur = a
        for i in range(1, n-1):
            exp = cur + d
            if arr[i] != exp: return int(exp)
            cur = exp
        
        return a